### DISCUSSION TITLE
Empty items array post cart transform

### DISCUSSION THREAD

**Discussion by stevekaran** [Shopify/function-examples#398] [Feedback: Cart Transform API]
When adding non-bundled products/variants to the cart via `/cart/add.js`, the response is a complete `items` array with a detailed JSON structure of the product/variant that was added to the cart, validating the item(s) were added successfully.

When adding child product variants of a bundle to the cart via `/cart/add.js`, the response returned is an empty items array `items: []`, instead of the complete `items` array like with non-bundled products. Only upon re-fetching the cart does one get the full JSON of the cart.

Is this intentional/as designed?

**Comment by ignacio-chiazzo** [comment]
Hi, that behaviour is unexpected. Could you provide more details so that we can investigate further? 

Note that the API will not return the bundle components as lines, instead the lines will be the "_parent_" bundles.

**Reply by stevekaran** [reply]
I am currently testing Cart Transform for my customer as they have a unique way a bundling products. I am adding 2x items to the cart at once as follows:

```javascript

fetch(Theme.routes.cart_add_url + ".js", { method: "POST", headers: { "Content-Type": "application/json", "X-Requested-With": "XMLHttpRequest" }, body: `{"items":[{"id":"45473501249852","quantity":"1"},{"id":"45473501282620","quantity":"1"}]}`})
``` 
Here are screenshots of the request and the response.
![image](https://github.com/Shopify/function-examples/assets/149213887/de5c97b9-78c8-4fc7-a7de-da78c6d61fd4)
![image](https://github.com/Shopify/function-examples/assets/149213887/076adca8-0217-4074-89ee-1a67c1d1ad5a)

Only after I re-fetch the cart does the response contain the details of the parent bundle product.
`fetch(Theme.routes.cart_url, { headers: { accept: "application/json" }})`
![image](https://github.com/Shopify/function-examples/assets/149213887/e7b7077c-7511-4403-b4d4-d8ae7ab08e24)

Cart JSON
```javascript
{
  "token": "c1-e8ef000c379c57be10ffd798020aa7bf",
  "note": "",
  "attributes": {},
  "original_total_price": 2499,
  "total_price": 2499,
  "total_discount": 0,
  "total_weight": 0.0,
  "item_count": 1,
  "items": [
    {
      "id": 47807403131196,
      "properties": {},
      "quantity": 1,
      "variant_id": 47807403131196,
      "key": "47807403131196:e0ea27f99c9b13d580949e2608414d83",
      "title": "Elite Donut Test Bundle",
      "price": 2499,
      "original_price": 2499,
      "discounted_price": 2499,
      "line_price": 2499,
      "original_line_price": 2499,
      "total_discount": 0,
      "discounts": [],
      "sku": "",
      "grams": 0,
      "vendor": "shdev2",
      "taxable": true,
      "product_id": 8861342105916,
      "product_has_only_default_variant": true,
      "gift_card": false,
      "final_price": 2499,
      "final_line_price": 2499,
      "url": "\/products\/elite-donut-test-bundle?variant=47807403131196",
      "featured_image": {
        "aspect_ratio": 1.0,
        "alt": "Elite Donut Test Bundle",
        "height": 600,
        "url": "https:\/\/cdn.shopify.com\/s\/files\/1\/0755\/9151\/6476\/files\/EliteSweetsProteinDonutsCinnamon6pk_c6c191de-31ef-4865-9016-87d4a92e0ef2.jpg?v=1697515565",
        "width": 600
      },
      "image": "https:\/\/cdn.shopify.com\/s\/files\/1\/0755\/9151\/6476\/files\/EliteSweetsProteinDonutsCinnamon6pk_c6c191de-31ef-4865-9016-87d4a92e0ef2.jpg?v=1697515565",
      "handle": "elite-donut-test-bundle",
      "requires_shipping": true,
      "product_type": "",
      "product_title": "Elite Donut Test Bundle",
      "product_description": "Elite Donut Test Bundle",
      "variant_title": null,
      "variant_options": [
        "Default Title"
      ],
      "options_with_values": [
        {
          "name": "Title",
          "value": "Default Title"
        }
      ],
      "line_level_discount_allocations": [],
      "line_level_total_discount": 0,
      "quantity_rule": {
        "min": 1,
        "max": null,
        "increment": 1
      },
      "has_components": true
    }
  ],
  "requires_shipping": true,
  "currency": "CAD",
  "items_subtotal_price": 2499,
  "cart_level_discount_applications": []
}
```

**Reply by stevekaran** [reply]
@ignacio-chiazzo Following up here... Did you need any additional info? Should I open a ticket with support, or is this forum ok?

**Reply by stevekaran** [reply]
Sorry. missed your update. I came back to this thread as the issue remains. This forum is fine, whatever process works best for you.

**Reply by stevekaran** [reply]
@ignacio-chiazzo following up on this thread. Any updates?

**Comment by stevekaran** [comment]
@ignacio-chiazzo @nickwesselman Opened a ticket with partner support for proper action/resolution

**Comment by SkylarRohter** [comment]
Any updates on this?
